include/aws/s3/s3_client.h

```diff
@@ -233,6 +283,15 @@ struct aws_s3_client_config {
     aws_s3_client_shutdown_complete_callback_fn *shutdown_callback;
     void *shutdown_callback_user_data;
 
+    /**
+     * Optional.
+     * To get telemetry metrics when a single request finishes.
+     * If set the request will keep track the metrics from `aws_s3_request_metrics`, and fire the callback when the
+     * request finishes receiving response.
+     */
+    aws_s3_meta_request_telemetry_fn *telemetry_callback;
+    void *telemetry_callback_user_data;
```

Comment:
I think it will be multiple callbacks like part start , part end etc. `telemetry_callback` is not a good name, maybe something like `on_requst_finished_callback`